src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -228,6 +228,10 @@ private X509Certificate2 EnsurePrivateKey(X509Certificate certificate)
                         certEx.Dispose();
                     }
                 }
+                else
+                {
+                    certHash = certificate.GetCertHashString();
```

Comment:
This change deviates from the .NET Framework behavior, because it would allow a 3rd party type derived from `X509Certificate` which returns IntPtr.Zero for the handle value to function as a lookup key.  NetFx seems to go to a lot of work to avoid that happening.

I'd probably wrap the rest of this try block in `if (certHash != null)` and leave this else out (@davidsh, your thoughts?).

Reply:
Yes you are right. I will make the necessary changes.